lib/cloud_controller/opi/apps_client.rb

```diff
@@ -8,252 +8,107 @@
-        cpu_weight: cpu_weight,
-        health_check_type: process.health_check_type,
-        health_check_http_endpoint: process.health_check_http_endpoint,
-        health_check_timeout_ms: timeout_ms,
-        start_timeout_ms: health_check_timeout_in_seconds(process) * 1000,
-        last_updated: process.updated_at.to_f.to_s,
-        volume_mounts: generate_volume_mounts(process),
-        ports: process.open_ports,
-        routes: routes(process),
-        lifecycle: lifecycle.to_hash,
-        user_defined_annotations: filter_annotations(process.app.annotations)
-      }
-      MultiJson.dump(body)
-    end
-
-    def filter_annotations(annotations)
-      annotations.select { |anno| is_prometheus?(anno)
-      }.map { |anno| ["#{anno.key_prefix}/#{anno.key}", anno.value] }.to_h
-    end
-
-    def is_prometheus?(anno)
-      !anno.key_prefix.nil? && anno.key_prefix.start_with?(PROMETHEUS_PREFIX)
-    end
-
-    def health_check_timeout_in_seconds(process)
-      process.health_check_timeout || config.get(:default_health_check_timeout)
-    end
-
-    def update_body(process)
-      body = {
-        guid: process.guid,
-        version: process.version,
-        update: {
-          instances: process.desired_instances,
-          routes: routes(process),
-          annotation: process.updated_at.to_f.to_s
-        }
-      }
-      MultiJson.dump(body)
-    end
-
-    def routes(process)
-      routing_info = VCAP::CloudController::Diego::Protocol::RoutingInfo.new(process).routing_info
-      http_routes = (routing_info['http_routes'] || []).map do |i|
+      lifecycle = OPI.lifecycle_for(process).to_hash
+      lrp = Kubeclient::Resource.new(
         {
-          hostname: i['hostname'],
-          port: i['port']
+          metadata: {
+            name: process.app.name,
+            namespace: @config.kubernetes_workloads_namespace
+          },
+          spec: {
+            GUID: process.guid,
+            version: process.version,
+            processType: process.type,
+            appGUID: process.app.guid,
+            appName: process.app.name,
+            spaceGUID: process.space.guid,
+            spaceName: process.space.name,
+            orgGUID: process.organization.guid,
+            orgName: process.organization.name,
+            command: lifecycle[:docker_lifecycle][:command],
+            image: lifecycle[:docker_lifecycle][:image],
+            env: OPI.hash_values_to_s(OPI.environment_variables(process)),
+            instances: process.desired_instances,
+            memoryMB: process.memory,
+            cpuWeight: cpu_weight,
+            diskMB: process.disk_quota,
+            health: {
+              type: process.health_check_type,
+              timeoutMs: timeout_ms
+            },
+            lastUpdated: process.updated_at.to_f.to_s,
+            volumeMounts: generate_volume_mounts(process),
+            ports: process.open_ports,
+            appRoutes: OPI.routes(process),
```

Comment:
Does this stay in sync as the user maps and unmaps routes for a running LRP? I believe the same code that does it for Diego LRPs should still be doing it in this case, but am wondering if you've tested this.

Reply:
It wouldn't update, since that requires the update to be implemented with CRs, which we'll do in a future PR. The `OPI.routes` function is just extracted in the OPI module, but still calls `VCAP::CloudController::Diego::Protocol::RoutingInfo.new(process).routing_info` to get the routes